php/default-formatting-tags.php

```diff
@@ -541,6 +541,36 @@ public function get_value() {
 }
 
 /**
+ * Formatting tag for the post thumbnail URL.
+ */
+class WP_SEO_Format_Thumbnail_URL extends WP_SEO_Formatting_Tag {
+	/**
+	 * Tag name.
+	 *
+	 * @var string
+	 */
+	public $tag = '#thumbnail_url#';
+
+	/**
+	 * Get the tag description.
+	 *
+	 * @return string
+	 */
+	public function get_description() {
+		return __( 'Replaced with the URL for the full-size featured image of the content being viewed.', 'wp-seo' );
+	}
+
+	/**
+	 * Get the tag value for the current page.
+	 *
+	 * @return mixed Thumbnail URL, or false.
+	 */
+	public function get_value() {
+		return ( is_singular() ) ? get_the_post_thumbnail_url( get_the_ID(), 'full' ) : false;
```

Comment:
I would filter `'full'`. I don't _love_ it as the default, but it's the only choice you know will always work. I could be convinced of `'large'` instead, and I could also be convinced to pass an array and let WordPress choose the best cut.